runtime/evaluate_benchmark.cc

```diff
@@ -0,0 +1,205 @@
+#include <benchmark/benchmark.h>
+
+#include <atomic>
+#include <cstddef>
+#include <cstdint>
+
+#include "runtime/evaluate.h"
+#include "runtime/expr.h"
+#include "runtime/thread_pool.h"
+
+namespace slinky {
+
+node_context ctx;
+var buf(ctx, "buf");
+var buf2(ctx, "buf2");
+var x(ctx, "x");
+var y(ctx, "y");
+var z(ctx, "z");
+
+constexpr index_t iterations = 100000;
+
+// These benchmarks mostly work by generating nodes around a call counter, and wrapping that node with a loop.
+stmt make_call_counter(std::atomic<int>& calls) {
+  return call_stmt::make(
+      [&](eval_context& ctx) -> index_t {
+        ++calls;
+        return 0;
+      },
+      {}, {});
+}
+
+stmt make_loop(stmt body) { return loop::make(x.sym(), loop_mode::serial, range(0, iterations), 1, body); }
+
+// For nodes that need a buffer, we can add a buffer outside that loop, the cost of constructing it will be negligible.
+stmt make_buf(int rank, stmt body) {
+  std::vector<dim_expr> dims;
+  for (int i = 0; i < rank; ++i) {
+    dims.push_back({{0, 100}, i, dim::unfolded});
+  }
+  return make_buffer::make(buf.sym(), 0, 1, dims, body);
+}
+
+void BM_call(benchmark::State& state) {
+  std::atomic<int> calls = 0;
+  stmt body = make_loop(make_call_counter(calls));
+
+  for (auto _ : state) {
+    evaluate(body);
+  }
+
+  state.SetItemsProcessed(calls);
+}
+
+BENCHMARK(BM_call);
+
+void BM_let(benchmark::State& state) {
```

Comment:
Would be interesting to add more benchmarks for let with multiple entries, to measure how much each additional let in the node costs